app/Http/Controllers/OAuthController.php

```diff
@@ -27,7 +27,13 @@ public function socialite(string $service): symfonyResponse
 
     public function callback(string $service): RedirectResponse|Redirector
     {
-        $authUser = Socialite::driver($service)->user();
+        // Try/Catch to prevent 500  error when access is denied at provider
+        try {
+            $authUser = Socialite::driver($service)->user();
+        } catch (\Exception $e) {
+            return redirect("/login");
```

Comment:
Would these errors be meaningful to an administrator who can look at the logs?  If so, you can use the [`report()`](https://laravel.com/docs/11.x/errors#the-report-helper) helper to automatically log the error without rendering an error page.

Reply:
It could be.  I came across it when clicking `no` on a GitLab login, which isn't something I would expect an admin to care about.  But if its happening for all users of a particular provider, then yes.   I guess it couldn't hurt to report it.

Reply:
After thinking about it more, I think it's probably fine as-is.  You could maybe add a log message about it, but I think putting entire stack traces in the log for something a user might legitimately do could be a bit much.

Reply:
> You could maybe add a log message about it

A good compromise.  I'll do that and resubmit.